tests/apps/qchem/test_dynamics.py

```diff
@@ -328,6 +234,135 @@ def test_negative_state(self, e):
             dynamics.prob(samples, [-i for i in state])
 
 
+@pytest.mark.parametrize("c", [c1, c2])
+class TestSampleCoherent:
+    """Tests for the function ``strawberryfields.apps.qchem.dynamics.sample_coherent``"""
+
+    def test_invalid_n_samples(self, c):
+        """Test if function raises a ``ValueError`` when a number of samples less than one is
+        requested."""
+        with pytest.raises(ValueError, match="Number of samples must be at least one"):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, t, U, w, 0)
+
+    def test_complex_unitary(self, c):
+        """Test if function raises a ``ValueError`` when a complex unitary is given."""
+        with pytest.raises(
+            ValueError, match="The normal mode to local mode transformation matrix must be real"
+        ):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, t, 1.0j * U, w, ns)
+
+    def test_invalid_mode(self, c):
```

Comment:
Thanks for moving some of the tests into the combined test class! I think some more tests, such as this one, could potentially go there. However, I think it's too much time vs reward, so let's not bother! :+1: